pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -59,38 +59,31 @@ func applyLocalityWeight(
 	// By setting weights across different localities, it can allow
 	// Envoy to weight assignments across different zones and geographical locations.
 	for _, localityWeightSetting := range distribute {
-		if localityWeightSetting != nil &&
-			util.LocalityMatch(locality, localityWeightSetting.From) {
-			misMatched := map[int]struct{}{}
-			for i := range loadAssignment.Endpoints {
-				misMatched[i] = struct{}{}
-			}
+		if localityWeightSetting != nil && util.LocalityMatch(locality, localityWeightSetting.From) {
+			everMatched := map[int]struct{}{}
 			for locality, weight := range localityWeightSetting.To {
-				// index -> original weight
-				destLocMap := map[int]uint32{}
-				totalWeight := uint32(0)
+				inLocality := map[int]struct{}{}
 				for i, ep := range loadAssignment.Endpoints {
-					if _, exist := misMatched[i]; exist {
-						if util.LocalityMatch(ep.Locality, locality) {
-							delete(misMatched, i)
-							destLocMap[i] = ep.LoadBalancingWeight.Value
```

Comment:
So I think we check the `LoadBalancingWeight` here can prevent panic

Reply:
I guess if we default to 1 here this would work without other changes, but I think it's unneeded complexity. It also makes me function not idempotent